libnmstate/route_rule.py

```diff
@@ -159,6 +199,8 @@ def gen_metadata(self, route_state, ifaces):
                 Interface.IPV4: [],
                 Interface.IPV6: [],
             }
+            if rules != self._cur_rules[route_table]:
+                ifaces[iface_name].mark_as_changed()
```

Comment:
That handles the case if all the rules has been removed from a table. The structure `self._rules` is {table: [rules]}, therefore when all the rules has been removed it will be `{table: []}` and the generated metadata will be {iface_name: {ipv4: [], ipv6: []}} which means that all the rules has been removed.

In addition, the code is comparing self._rules with self._cur_rules for that table, so if it changed the interface will be marked as changed.